### ISSUE TITLE
Red screen on launch

### ISSUE THREAD

**Issue by camelid** [servo/servo#26356]
For every site, when I load it, a red screen is shown for several seconds. (Same with local test HTML files.)

Here's what it looks like:

<img width="1016" alt="servo red screen on startup" src="https://user-images.githubusercontent.com/37223377/80645804-7f723880-8a20-11ea-8671-803e17aad8a6.png">

@jdm thinks this is because of #25853 and at some point the OpenGL surface is cleared to red.

**Comment by camelid** [comment]
Could this line where we set `clear_color` in `RendererOptions` be related to the issue? It's set to `None`, and perhaps when it's later used the default in OpenGL is red.

https://github.com/servo/servo/blob/343f249c1d1dbd46d0712e34c1baa9aafe19a75a/components/servo/lib.rs#L435

**Comment by camelid** [comment]
Nope. I tried changing it to `ColorF::WHITE`, but I still got the red screen.

**Comment by camelid** [comment]
@jdm Do you think this is an issue with Servo or with WebRender?

**Comment by jdm** [comment]
I'm pretty sure it's an issue with Servo.

**Comment by camelid** [comment]
Okay, thanks. I've been looking through the code in this repo and in surfman's. Do you think you could point me to the spot(s) in Servo where surfaces are created? Thanks!

**Comment by jdm** [comment]
https://github.com/servo/servo/blob/master/components/webrender_surfman/lib.rs#L56 is called from https://github.com/servo/servo/blob/343f249c1d1dbd46d0712e34c1baa9aafe19a75a/ports/glutin/headed_window.rs#L147-L158.

**Comment by camelid** [comment]
Cool, thanks!

**Comment by Darkspirit** [comment]
Maybe related to #26353.

**Comment by camelid** [comment]
Although it seems in that one that nothing ever rendered, whereas this issue is that on startup there's a red screen and then the page is rendered.

**Comment by camelid** [comment]
Tried adding the code from

https://github.com/servo/servo/blame/9dbc6554f087ca3675104fb1bac45b0c442a0158/ports/glutin/headed_window.rs#L191-L193

to

https://github.com/servo/servo/blob/d08c4fff15cb5f6a8840621e848c9de22fcc3439/components/servo/lib.rs#L374

at @asajeffrey's recommendation, but still the red screen.

**Comment by camelid** [comment]
@jdm is this an issue only on macOS or on Linux/Windows too?

**Comment by atouchet** [comment]
I do not see this issue using Windows 10.

**Comment by camelid** [comment]
Huh, interesting. Thanks for the info!

@asajeffrey Perhaps this is an issue with surfman's platform-specific APIs then?

**Comment by camelid** [comment]
I just tried a prebuilt Servo binary on my Ubuntu VM and it worked fine without the red screen.

**Comment by camelid** [comment]
Likely an issue with the platform specific background code, e.g. surfman, winit. Needs more experimentation.

**Comment by camelid** [comment]
I tried running the `chaos_game` example in surfman, but no red screen - it worked fine.

I'm guessing the red screen is either caused by winit or the way servo calls winit or surfman.

**Comment by jdm** [comment]
We haven't had any further reports of this in years, so closing optimistically.

**Comment by camelid** [comment]
Nice to see an update on this issue, even if it's an optimistic close. :)
